subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -1170,10 +1170,52 @@ BT_GATT_SERVICE_DEFINE(bass_svc,
 	RECEIVE_STATE_CHARACTERISTIC(2)
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 2 */
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 1 */
-);
+};
+
+static struct bt_gatt_service bass_svc = BT_GATT_SERVICE(attr_bass_svc);
+
+static bool scan_delegator_registered;
+static bool pa_sync_cb_registered;
+
+static int bt_bap_bass_register(void)
+{
+	int err;
+
+	if (bass_svc.attr_count == 0) {
+		LOG_ERR("BASS service not initialized properly");
```

Comment:
Since this initialization happens statically, this shouldn't be able to happen. On this premise, IF this happens something is very wrong. Consider whether we should use an ASSERT here

Reply:
Good point...

Reply:
Since bass_svc is not a pointer this cannot fail. However, the same is true even if you do &bass_svc, as its lifetime always will surpass that of the funciton scope.

I think that we should check what you initially had, e.g. bass_svc.attr_count != 0. I think it would also be fine with just removing this check altogether though, since bass_svc never is un/reassigned. The advantage of keeping it would ofc. be that it will guard against future changes that reassigns bass_svc - I haven no idea how likely that would be. And perhaps such a potential change should have responsibility of asserting a good value too.

Reply:
True, if we need to check it, we can either do it as before or use __ASSERT for 'attr_bass_svc'. However, I agree with you that there’s no need to check it since it is initialized right before this function. So, I’ll remove it.